bot/exts/filters/filtering.py

```diff
@@ -649,6 +651,12 @@ async def _has_rich_embed(msg: Message) -> Union[bool, List[Embed]]:
             for embed in msg.embeds:
                 if embed.type == "rich":
                     urls = URL_RE.findall(msg.content)
+                    final_urls = set(urls)
+                    # This is due to way discord renders relative urls in Embdes
+                    # if we send the following url: https://mobile.twitter.com/something
+                    # Discord renders it as https://twitter.com/something
+                    for url in urls:
+                        final_urls.add(remove_subdomain_from_url(url))
                     if not embed.url or embed.url not in urls:
```

Comment:
You've created a variable `final_urls` and are adding urls to it, but then the check is still `if ... embed.url not in urls`, so does this even do anything?

Also, we have the tldextract lib installed in the project, which is better at separating subddomains from urls than urllib and then assuming the first 2 parts are the subdomain (which isn't always the case)/.

Reply:
Shoot. The first implementation wasn't in a set and I was adding them in the original list, but I wanted to remove dups.
My bad, I'll fix that.

As for the `tldextract` one, i'll have a look at it right away